napari/util/misc.py

```diff
@@ -246,12 +246,17 @@ def calc_data_range(data):
     values : list of float
         Range of values.
     """
-    min = data.min()
-    max = data.max()
-    if min == max:
-        min = 0
-        max = 1
-    return [float(min), float(max)]
+    if np.prod(data.shape) > 1e6:
+        reduced_data = np.random.choice(data.flatten(), size=1e6)
+    else:
+        reduced_data = data
```

Comment:
Not in love with this change... Accessing 1M random points from a dask array could be extremely slow depending on backing (e.g. remote data). I still prefer dtype, but if you don't want that, I suggest taking e.g. the middle plane.

Reply:
yup, it's a concern for me too, if each of those is in a different chunk .... I just wasnt quite sure what to do. How about top, middle, and bottom planes? Seems like an ok compromise for now!!